File: tesis-slm-quechua/apps/pwa/app/rag/constructorContexto.ts
// Puerto literal de src/rag/constructor_contexto.py

import { ChunkConScore } from './indiceBm25';

export function buildContext(chunks: ChunkConScore[], maxChunks = 5): string {
  const parts: string[] = [];
  chunks.slice(0, maxChunks).forEach((c, i) => {
    const sid = c.source_id ?? '?';
    const ref = c.source_ref ?? '';
    const text = (c.text ?? '').trim();
    parts.push(`[${i + 1}] source=${sid} ref=${ref}\n${text}`);
  });
  return parts.join('\n\n');
}

File: tesis-slm-quechua/src/rag/constructor_contexto.py
from __future__ import annotations

from typing import Any


def build_context(chunks: list[dict[str, Any]], max_chunks: int = 5) -> str:
    parts = []
    for i, c in enumerate(chunks[:max_chunks], start=1):
        sid = c.get("source_id", "?")
        ref = c.get("source_ref", "")
        text = c.get("text", "").strip()
        parts.append(f"[{i}] source={sid} ref={ref}\n{text}")
    return "\n\n".join(parts)


construir_contexto = build_context

